cloudinit/distros/gentoo.py

```diff
@@ -27,28 +28,31 @@ class Distro(distros.Distro):
 
     def __init__(self, name, cfg, paths):
         distros.Distro.__init__(self, name, cfg, paths)
+
+        if distros.uses_systemd():
+            self.hostname_conf_fn = "/etc/hostname"
+        else:
+            self.hostname_conf_fn = "/etc/conf.d/hostname"
+
         # This will be used to restrict certain
         # calls from repeatedly happening (when they
         # should only happen say once per instance...)
         self._runner = helpers.Runners(paths)
         self.osfamily = "gentoo"
-        # Fix sshd restarts
-        cfg["ssh_svcname"] = "/etc/init.d/sshd"
-        if distros.uses_systemd():
-            LOG.error("Cloud-init does not support systemd with gentoo")
+        if not distros.uses_systemd():
+            # Fix sshd restarts (openrc-specific?)
+            cfg["ssh_svcname"] = "/etc/init.d/sshd"
```

Comment:
The default value is `sshd`. This is used to restart sshd if it is already running so that user configuration can be applied.

Reply:
I mainly kept this because I didn't want to touch the openrc behaviour at all (and the code was there already).
Was added in 2014, commit bc77232588dd587849e8b3b7fc599387b6c905fa - and does not make much sense to me either.

Reply:
OK, after looking at alpine (which just sets 'sshd' - the default), this block should be safe to drop. Adding a commit.